src/main/java/ua/realestate/management/entity/User.java

```diff
@@ -29,31 +27,15 @@
     private String password;
     private Boolean isActive;
 
+    @OneToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "user_roles")
+    @NonNull
+    private UserRole userRole;
 
-    @ElementCollection(targetClass = Role.class, fetch = FetchType.EAGER)
-    @CollectionTable(name = "user_roles", joinColumns = @JoinColumn(name = "user_id"))
-    @Enumerated(EnumType.STRING)
-    private Set<Role> roles;
-
-//    public String getName() {
-//        return name;
-//    }
-//
-//    public void setName(String name) {
-//        this.name = name;
-//    }
-
-    public Set<Role> getRoles() {
-        return roles;
-    }
-
-    public void setRoles(Set<Role> roles) {
-        this.roles = roles;
-    }
 
     @Override
     public Collection<? extends GrantedAuthority> getAuthorities() {
-        return getRoles();
+        return null;
```

Comment:
be careful with implementing any methods and always read java doc:
```Returns the authorities granted to the user. Cannot return <code>null</code>.```

Reply:
fixed